docs/websites.md

```diff
@@ -0,0 +1,206 @@
+
+You can either [setup a custom domain to point to this URL](environment/custom-domains.md#custom-domains-for-static-websites-on-s3) or setup CloudFront as explained below.
+
+### Uploading files to S3
+
+It is not possible to use `sam deploy` to upload files to S3, you need to upload them separately. To do this, you can use the `aws s3 sync` command:
+
+```bash
+aws s3 sync <directory> s3://<bucket-name> --delete
+```
+
+Be aware that the content of the bucket is public!
+
+## Serving PHP and static files via CloudFront
+
+This section assumes you have already deployed your static files on S3 as shown above.
+
+As explained at the beginning of this page, CloudFront will let us:
+
+- serve the PHP application and static files under the same domain
+- support both HTTP and HTTPS
+
+This diagram helps understand how CloudFront works:
+
+![](cloudfront.png)
+
+CloudFront dispatches HTTP requests to "Origins" (API Gateway/Lambda, S3, etc.) based on "Cache Behaviors". A cache behavior can match a specific URL pattern, and can have a specific caching configuration (e.g. cache assets for a long time but do not cache PHP responses).
+
+The `template.yaml` example below:
+
+- receives HTTP and HTTPS requests
+- serves everything URL that starts with `/assets/` using S3 (static files)
+- sends all the other requests to PHP via API Gateway
+
+Feel free to customize the `/asset/` path. If your application is a JavaScript application backed by a PHP API, you will want to invert API Gateway and S3 (set S3 as the `DefaultCacheBehavior` and serve API Gateway under a `/api/` path).
+
+```yaml
+Resources:
+    ...
+    WebsiteCDN:
+        Type: AWS::CloudFront::Distribution
+        Properties:
+            DistributionConfig:
+                Enabled: true
+                # Cheapest option by default (https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_DistributionConfig.html)
+                PriceClass: PriceClass_100
+                # Origins are where CloudFront fetches content
+                Origins:
+                    # The website (AWS Lambda)
+                    -   Id: Website
+                        DomainName: !Sub '${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com'
+                        OriginPath: '/Prod'
+                        CustomOriginConfig:
+                            OriginProtocolPolicy: 'https-only' # API Gateway only supports HTTPS
+                    # The assets (S3)
+                    -   Id: Assets
+                        # Watch out, use s3-website URL (https://stackoverflow.com/questions/15309113/amazon-cloudfront-doesnt-respect-my-s3-website-buckets-index-html-rules#15528757)
+                        DomainName: !Sub '${Assets}.s3-website-${AWS::Region}.amazonaws.com'
```

Comment:
Should also add a note about the structure of the s3 url. https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints

Reply:
Thanks to #324 we can simplify the URL to `https://<bucket>.s3.amazonaws.com/<file>`